mkosi/__init__.py

```diff
@@ -258,6 +262,13 @@ def configure_os_release(context: Context) -> None:
                 elif context.config.image_version and line.startswith("IMAGE_VERSION="):
                     new.write(f'IMAGE_VERSION="{context.config.image_version}"\n')
                     image_version_written = True
+                elif (
+                    context.config.distribution == Distribution.debian and
+                    context.config.release in ("unstable", "sid") and
+                    line.startswith("VERSION_CODENAME=")
+                ):
+                    # Workaround for Debian brokeness: unstable sets VERSION_CODENAME=<testing codename>
+                    new.write('VERSION_CODENAME=sid\n')
```

Comment:
Please move this to the debian installer class `install()` function. Also add a link to upstream bug report please, I don't like carrying distribution workarounds forever.

Reply:
Ok, but I'll need to completely duplicate the function then?

Reply:
Yeah that's fine, just duplicate it but without all the image ID stuff

Reply:
Well it would need that too? Because of the diversion

Reply:
Is the cache discarded if the image id or version change? Because by moving it there, it will only run when the cache is discarded, as opposed to now where it always runs

Reply:
looks like it is